### PR TITLE
[BUILD] Build multi-platform (amd64/arm64) for docker image by JIB - Fix Jenkinsfile dir

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -53,7 +53,9 @@ pipeline {
 
               echo "Docker tag: ${env.DOCKER_TAG}"
               // build and push docker images
-              sh "mvn -Pci jib:build -pl apps/distributed,apps/distributed-es6-backport,apps/memory -X"
+                dir("tmail-backend") {
```
Comment (Arsnael) [inline Jenkinsfile:41]: indent?

**Comment by vttranlina** [comment]
I get the issue with the docker hub credential (when push image)
I tried to get it from env
`-Djib.to.auth.username=${env.DOCKER_HUB_USER} -Djib.to.auth.password=${env.DOCKER_HUB_TOKEN}`
but the environment var is only available on the Jenkins master node, It does not share to slave.

//
The old way the pushing was done by Jenkin docker plugin?
The new way the pushing is done by JIB https://github.com/GoogleContainerTools/jib/blob/master/jib-maven-plugin/README.md#using-specific-credentials

**Review by Arsnael — COMMENTED** [review]
> but the environment var is only available on the Jenkins master node, It does not share to slave.

Hmmmm.... Maybe we could try to check if we can add those env variable into the slave containers when starting them?

**Review by Arsnael — APPROVED** [review]
It seems to work :)

You are lucky i was about to revert the last commit => https://github.com/linagora/tmail-backend/pull/875

File: Jenkinsfile [hunk 2]
```diff
@@ -24,26 +28,11 @@ pipeline {
...
-                }
-            }
-            post {
-                always {
-                    junit(testResults: '**/surefire-reports/*.xml', allowEmptyResults: false)
-                }
-                failure {
-                    archiveArtifacts artifacts: '**/target/test-run.log' , fingerprint: true
-                    archiveArtifacts artifacts: '**/surefire-reports/*' , fingerprint: true
+                    sh 'echo "TODO: add tests"'
```
Comment (quantranhong1999) [inline Jenkinsfile:31]: debug

File: Jenkinsfile [hunk 3]
```diff
@@ -24,26 +28,11 @@ pipeline {
...
                 }
             }
         }
         stage('Deliver Docker images') {
-          when {
-            anyOf {
-              branch 'master'
-              buildingTag()
-            }
-          }
```
Comment (quantranhong1999) [inline Jenkinsfile:46]: Should keep this at the end.

File: tmail-backend/apps/distributed-es6-backport/pom.xml [hunk 4]
```diff
@@ -524,12 +524,8 @@
                             <to>
                                 <image>linagora/tmail-backend</image>
                                 <tags>
-                                    <tag>distributed-esv6-${env.DOCKER_TAG}</tag>
+                                    <tag>distributed-esv6-branch-master</tag>
```
Comment (quantranhong1999) [inline tmail-backend/apps/distributed-es6-backport/pom.xml:527]: Should keep this at the end. As we have tag when release, not only master tag.

File: Jenkinsfile [hunk 5]
```diff
@@ -24,26 +28,11 @@ pipeline {
...
-                    sh 'mvn -B surefire:test'
-                }
-            }
-            post {
-                always {
-                    junit(testResults: '**/surefire-reports/*.xml', allowEmptyResults: false)
-                }
-                failure {
-                    archiveArtifacts artifacts: '**/target/test-run.log' , fingerprint: true
-                    archiveArtifacts artifacts: '**/surefire-reports/*' , fingerprint: true
```
Comment (quantranhong1999) [inline Jenkinsfile:36]: Should keep test and test artifact at the end.

**Review by quantranhong1999 — COMMENTED** [review]
Seems Tung is testing without the maven test phase for faster. We should not merge this yet.

**Comment by Arsnael** [comment]
Well caught @quantranhong1999 , I just focused in the solution in the end, forgot about the bypasses. My mistake :)

File: tmail-backend/apps/distributed-es6-backport/pom.xml (hunk 4 above)
Comment (vttranlina) [reply tmail-backend/apps/distributed-es6-backport/pom.xml]: I Override custom tag in Jenkinsfile, see it in the last commit

**Comment by vttranlina** [comment]
For issue: https://github.com/linagora/tmail-backend/pull/874#issuecomment-1851174540

I extracted the credentials by syntax

```
    environment {
        DOCKER_HUB_CREDENTIAL = credentials('dockerHub')
    }
```
So, we don't need to set env variable for all workers.

**Review by Arsnael — APPROVED** [review]
LGTM :) 

It looks like I don't need to care of esv6 version being removed anymore, as you do it here :)

**Pull request merged**